### DISCUSSION TITLE
Solved - Lesson 4 contract deployment reverted

### DISCUSSION THREAD

**Discussion by RichardPalladino** [Cyfrin/foundry-full-course-cu#143] [Q&A] [answered]
I'm writing this in case in case someone else encounters the same issue I just resolved.

My conditions are:

1. I'm coding and deploying alongside the video around Lesson 4;
2. I have nearly identical code (having pulled out some code to turn it into a library) and my code compiles;
3. I have more than enough Sepolia testnet ETH and the environment is set to injected provider; and
4. Every time I try to deploy the contract, I am told the transaction will revert with a generic reversion error.

I also have multiple workspaces in Remix because I have used it for over a year to learn and prototype things.

Solution:

- I simply navigated away from my Lesson 4 workspace to do something else and magically the exact same, unchanged contract deploys without giving a reversion error after I return to the workspace to try again.

In other words, there appears to be a bug with Remix where it throws a reversion error occasionally.  The fix is to navigate away from the workspace and then back to try again.  I presume this is some kind of caching error or similar as a result of multiple workspace environments.

**Comment by RichardPalladino** [comment, marked as answer]
I'm going to add that I'm experiencing these random reversion errors when interacting with the testnet through Remix often.

For example, I received the reversion error from Remix after I funded the contract and played with it a little when I was finally ready to withdraw.

I copied the deployed contract address, navigated away to a random other workspace, navigated back to my lesson 4 workspace, recompiled FundeMe.sol, selected the compiled contract in "Deploy and Run Transactions", pasted the already deployed contract's address next to "At Address" and clicked the "At Address" button.  

Withdraw() function then worked without error or reversion.  

Nothing changed.  I just navigated away from the workspace in Remix and then back and Remix no longer threw a reversion error when I tried the exact same call.
